slothy/targets/aarch64/aarch64_neon.py

```diff
@@ -250,28 +227,63 @@ def extract(source, lbl):
                     pre.append(l)
                 continue
             if state == 1:
-                p = loop_end_regexp[0].match(l_str)
+                p = loop_end_regexp[loop_end_ctr].match(l_str)
                 if p is not None:
-                    reg0 = p.group("reg0")
-                    reg1 = p.group("reg1")
-                    imm = p.group("imm")
-                    state = 2
+                    if additional_data is None:
```

Comment:
Can you say what the intended logic is here? Are you expecting this to be set by the first end-regexp?

Reply:
Yes, the first end-regexp sets `additional_data`. I made this choice because that's usually the `subs` or `cmp` in which we can see the counter register and/or decrement. 
I realize that maybe I want to get `additional_data` for each instruction in the "end"-part and then merge the dictionaries in the end. I'll change that.